src/app/ui/editor/pixels_movement.cpp

```diff
@@ -791,27 +850,78 @@ void PixelsMovement::stampImage(bool finalStamp)
     stampExtraCelImage();
   }
 
+  // Saving original values before the 'for' loop and
+  // 'reproduceAllTransformationsWithInnerCmds' for restoring later.
+  // All these values will changed during the transformation reproduction.
+  TilemapMode originalSiteTilemapMode = (m_site.tilemapMode() == TilemapMode::Tiles &&
+                                         m_site.layer()->isTilemap()? TilemapMode::Tiles : TilemapMode::Pixels);
+  TilesetMode originalSiteTilesetMode = m_site.tilesetMode();
+  std::unique_ptr<Mask> initialMask0(new Mask(*m_initialMask0.get()));
+  std::unique_ptr<Mask> initialMask(new Mask(*m_initialMask.get()));
+  std::unique_ptr<Mask> currentMask(new Mask(*m_currentMask.get()));
+  std::unique_ptr<Transformation> initialData(new Transformation(m_initialData));
+  std::unique_ptr<Transformation> currentData(new Transformation(m_currentData));
+
```

Comment:
I think we can now use std::make_unique<T>(), instead of the new operator. In fact there are places ni the code where it was replaced already.